docs/generating-tests.rst

```diff
@@ -0,0 +1,370 @@
+
+Goal here is the get the ``Aleth`` Travis CI build to run the new tests with ``Aleth`` to check they pass. To do that a PR has to be submitted to Aleth that updates the git submodule for ethereum/tests to point to a branch with the new tests.
+
+Preparations on the ethereum/tests side
+---------------------------------------
+
+For trying the filled test(s) on ``Travis CI`` for ``Aleth``, the new test cases need to exist in a branch in ``ethereum/tests``. For this, ask somebody with a push permission to ``ethereum/tests``.
+
+
+Preparations on the Aleth side
+------------------------------
+
+Enter ``aleth/test/jsontests`` directory, and checkout the new branch in ``ethereum/tests`` as described in the instructions above. Then go back to the main ``Aleth`` directory and perform ``git add test/jsontests`` followed by ``git commit``.
+
+When you file this commit as a pull request to ``Aleth``, Travis CI should try the newly filled tests.
+
+git commit
+----------
+
+After these are successful, the filler file and the filled test should be added to the ``tests`` repository. File these as a pull request.
+
+If changes in the ``Aleth`` code itself were necessary, also file a pull request for these changes.
+
+Advanced: Converting a GeneralStateTest Case into a BlockchainTest Case
+=======================================================================
+
+In the tests repository, each GeneralStateTest is eventually translated into a BlockchainTest.  This can be done by the following sequence of commands.
+
+.. code::
+
+  ETHEREUM_TEST_PATH="../../tests" test/testeth -t GeneralStateTests/stExample2 -- --filltests --fillchain
+
+
+followed by
+
+.. code::
+
+  ETHEREUM_TEST_PATH="../../tests" test/testeth -t GeneralStateTests/stExample2 -- --filltests
+
+
+The second command is necessary because the first command modifies the GeneralStateTests in an undesired way.
+
+After these two commands,
+
+
+* ``git status`` to check if any GeneralStateTest has changed.  If yes, revert the changes, and follow section _\ ``Trying the Filled Test Locally``.  That will probably reveail an error that you need to debug.
+* ``git add`` to add only the desired BlockchainTests.  Not all modified BlockchainTests are valuable because, when you run ``--fillchain`` twice, the two invocations always produce different BlockchainTests even there are no changes in the source.
+
+Advanced: When testeth Takes Too Much Time
+==========================================
+
+Sometimes, especially when you are running BlockchainTests, ``testeth`` takes a lot of time.
+
+This happens when the GeneralTest fillers contain wrong parameters.
+The ``"env"`` field should contain:
+
+.. code::
+
+     "currentCoinbase" : <an address>,
+     "currentDifficulty" : "0x020000",
+     "currentGasLimit" : <anything < 2**63-1 but make sure the transaction does not hit>,
+     "currentNumber" : "1",
```

Comment:
for generating blockchain tests version this field must be equal to 1 and timestamp to 1000
actually env might become depricated in state test fillers.

Reply:
Ok, have added both notes in appropriate places in the docs.